src/components/map/RouteLayer.js

```diff
@@ -184,9 +183,9 @@ class RouteLayer extends Component {
 
     let hfp = [];
 
-    if (selectedStop) {
-      const selectedStopObj = stops.find((s) => s.nodeId === selectedStop);
+    const selectedStopObj = stops.find((s) => s.nodeId === selectedStop);
 
+    if (selectedStopObj) {
```

Comment:
Just need to find the stop object in the array, and check if THAT is truthy. Since the stops array changes based on which route is selected, the queried stop may not actually have an object there, causing the app to crash.